src/Sylius/Bundle/ApiBundle/Serializer/CommandFieldItemIriToIdentifierDenormalizer.php

```diff
@@ -0,0 +1,103 @@
+use Sylius\Bundle\ApiBundle\Converter\ItemIriToIdentifierConverter;
+use Sylius\Bundle\ApiBundle\DataTransformer\CommandAwareInputDataTransformer;
+use Symfony\Component\DependencyInjection\ServiceLocator;
+use Symfony\Component\Serializer\Normalizer\ContextAwareDenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+final class CommandFieldItemIriToIdentifierDenormalizer implements ContextAwareDenormalizerInterface
+{
+    /** @var DenormalizerInterface */
+    private $objectNormalizer;
+
+    /** @var ItemIriToIdentifierConverter */
+    private $itemIriToIdentifierConverter;
+
+    /** @var CommandAwareInputDataTransformer */
+    private $commandAwareInputDataTransformer;
+
+    /** @var ServiceLocator */
+    private $commandFieldIriToIdentifierStorages;
+
+    public function __construct(
+        DenormalizerInterface $objectNormalizer,
+        ItemIriToIdentifierConverter $itemIriToIdentifierConverter,
+        CommandAwareInputDataTransformer $commandAwareInputDataTransformer,
+        ServiceLocator $commandFieldIriToIdentifierStorages
+    ) {
+        $this->objectNormalizer = $objectNormalizer;
+        $this->itemIriToIdentifierConverter = $itemIriToIdentifierConverter;
+        $this->commandAwareInputDataTransformer = $commandAwareInputDataTransformer;
+        $this->commandFieldIriToIdentifierStorages = $commandFieldIriToIdentifierStorages;
+    }
+
+    public function supportsDenormalization($data, $type, $format = null, array $context = [])
+    {
+        /** @var string $commandFieldItemIriToIdentifierStorageName */
+        foreach ($this->getCommandFieldIriToIdStoragesNames() as $commandFieldItemIriToIdentifierStorageName) {
+            /** @var CommandFieldItemIriToIdentifierStorageInterface $commandFieldItemIriToIdentifierConverter */
+            $commandFieldItemIriToIdentifierConverter = $this
+                ->commandFieldIriToIdentifierStorages
+                ->get($commandFieldItemIriToIdentifierStorageName)
+            ;
+
+            if ($this->getInputClass($context) === $commandFieldItemIriToIdentifierConverter->getCommandType()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public function denormalize($data, $type, $format = null, array $context = [])
+    {
+        $inputClass = $this->getInputClass($context);
+
+        /** @var string $commandFieldIriToIdentifierStorageName */
+        foreach ($this->getCommandFieldIriToIdStoragesNames() as $commandFieldIriToIdentifierStorageName) {
+            /** @var CommandFieldItemIriToIdentifierStorageInterface $commandFieldItemIriToIdentifierStorage */
+            $commandFieldItemIriToIdentifierStorage = $this
+                ->commandFieldIriToIdentifierStorages
+                ->get($commandFieldIriToIdentifierStorageName)
+            ;
+
+            if ($inputClass === $commandFieldItemIriToIdentifierStorage->getCommandType()) {
```

Comment:
This logic looks duplicated (look for the comment above) - I'd wrap it at least in a private method `provideCommandFieldName(string $inputClass): string` that throws an exception if it can't be found.